Automattic/wp-calypso
# Activity Log: Use feature checks rather than plan checks

#### Changes proposed in this Pull Request

* Replaced uses of `siteHasBackupProductPurchase` with feature checks.

#### Testing instructions



* Navigate to /activity-log with sites on various plans and make sure the output is the same as in production:
* Free site: <img width="1511" alt="image" src="https://user-images.githubusercontent.com/1398304/168904671-53e5ad18-f6fd-4adb-87c0-47bb832cfd12.png">
* Personal and higher: <img width="1511" alt="image" src="https://user-images.githubusercontent.com/1398304/168904925-e4252aa7-b1d1-44bb-9944-e0be34c83eec.png">
* Free Jetpack site in cloud: <img width="1511" alt="Screen Shot 2022-05-17 at 1 48 18 PM" src="https://user-images.githubusercontent.com/1398304/168907622-ffbf9ff2-5a3a-4ff8-a161-610e6149641b.png">
* Jetpack Site with Backup in cloud: 
<img width="1511" alt="image" src="https://user-images.githubusercontent.com/1398304/168907549-7fbe6b3a-b0fe-4aca-92a6-e83241a0be16.png">




Related to p4TIVU-a66-p2
